pkg/infrastructure/powervs/clusterapi/powervs.go

```diff
@@ -49,9 +50,13 @@ func leftInContext(ctx context.Context) time.Duration {
 	return duration
 }
 
+const privatePrefix = "api-int."
+const publicPrefix = "api."
+
 // InfraReady is called once cluster.Status.InfrastructureReady
 // is true, typically after load balancers have been provisioned. It can be used
 // to create DNS records.
+// nolint:gocyclo
```

Comment:
Maybe the linter has a point here and it's time to start breaking this functionality down into separate functions.

Reply:
yep I'm working on it

Reply:
@r4f4 are you okay with me refactoring in this PR?

Reply:
@mjturek sure. Just please break it down into commits to make the life of a fellow reviewer a bit easier :)
I also wouldn't mind if you add the `//nolint:` now and do the break down in a follow-up (as long as it's done).

Reply:
sounds good, let me address your other comments and I'll finish up the break down in a follow up patch this week that removes the `//nolint:`

Reply:
Cleanup patch is here https://github.com/openshift/installer/pull/8337 it's based on this one so theoretically we could merge it in one go. Up to you!